### ISSUE TITLE
Panic on deletion of private channels

### ISSUE THREAD

**Issue by mqp** [serenity-rs/serenity#655]
If you attempt to delete a private channel, you get sent a channel delete event for it, which triggers an [unreachable branch](https://github.com/serenity-rs/serenity/blob/current/src/model/event.rs#L156). Tested with serenity 0.6.2. (Continued from https://github.com/serenity-rs/serenity/issues/654#issue-467615752.)

To reproduce, just get a reference to a private channel, and try to delete it:

```rust
use serenity::client::Client;
use serenity::model::channel::PrivateChannel;
use serenity::prelude::{EventHandler, Context, RwLock};
use std::env;
use std::sync::Arc;

struct Handler;

impl EventHandler for Handler {
    fn private_channel_create(&self, ctx: Context, chan: Arc<RwLock<PrivateChannel>>) {
        chan.read().delete(&ctx).expect("I'm already dead so this expect doesn't matter");
    }
}

fn main() {
    let mut client = Client::new(&env::var("DISCORD_TOKEN").expect("token"), Handler).expect("Error creating client");
    if let Err(why) = client.start() {
        println!("An error occurred while running the client: {:?}", why);
    }
}
```

Result after messaging the bot (and thereby creating a private channel):
```
thread '<unnamed>' panicked at 'internal error: entered unreachable code', /.../serenity-0.6.2/src/model/event.rs:157:43
```

**Comment by Erk-** [comment]
Backtrace for the panic:
```
thread '<unnamed>' panicked at 'internal error: entered unreachable code', /home/erk/dev/serenity/src/model/event.rs:157:43
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:59
             at src/libstd/panicking.rs:197
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:211
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:474
   5: std::panicking::begin_panic
             at /rustc/a53f9df32fbb0b5f4382caaad8f1a46f36ea887c/src/libstd/panicking.rs:408
   6: <serenity::model::event::ChannelDeleteEvent as serenity::cache::cache_update::CacheUpdate>::update
             at /home/erk/dev/serenity/src/model/event.rs:157
   7: serenity::cache::Cache::update
             at /home/erk/dev/serenity/src/cache/mod.rs:825
   8: serenity::client::dispatch::update
             at /home/erk/dev/serenity/src/client/dispatch.rs:47
   9: serenity::client::dispatch::handle_event
             at /home/erk/dev/serenity/src/client/dispatch.rs:372
  10: serenity::client::dispatch::dispatch
             at /home/erk/dev/serenity/src/client/dispatch.rs:148
  11: serenity::client::bridge::gateway::shard_runner::ShardRunner<H,RH>::dispatch
             at /home/erk/dev/serenity/src/client/bridge/gateway/shard_runner.rs:218
  12: serenity::client::bridge::gateway::shard_runner::ShardRunner<H,RH>::run
             at /home/erk/dev/serenity/src/client/bridge/gateway/shard_runner.rs:154
  13: serenity::client::bridge::gateway::shard_queuer::ShardQueuer<H,RH>::start::{{closure}}
             at /home/erk/dev/serenity/src/client/bridge/gateway/shard_queuer.rs:211
```

**Comment by arqunis** [comment]
This should be fixed as of the commit [c:67f5e3d](https://github.com/serenity-rs/serenity/commit/67f5e3d970e23da320c091b19fc90314e15db83a).
